sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/ContainerRegistryClientBuilder.java

```diff
@@ -84,65 +114,90 @@ public ContainerRegistryClientBuilder credential(TokenCredential credential) {
     }
 
     /**
-     * Sets The HTTP httpPipeline to send requests through.
+     * Sets the HTTP pipeline to use for the service client.
+     * <p>
+     * If {@code pipeline} is set, all settings other than
+     * {@link #endpoint(String) endpoint} are ignored
+     * to build {@link ContainerRegistryAsyncClient} or {@link ContainerRegistryClient}.<br>
+     * </p>
+     *
+     * This service takes dependency on an internal policy which converts Azure token credentials into Azure Container Registry specific service credentials.
+     * In case you use your own pipeline you will have to create your own credential policy.<br>
      *
-     * @param httpPipeline the httpPipeline value.
-     * @return the ContainerRegistryBuilder.
+     * {For more information please see <a href="https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md"> Azure Container Registry Authentication </a> }.
+     *
+     * @param httpPipeline The HTTP pipeline to use for sending service requests and receiving responses.
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder pipeline(HttpPipeline httpPipeline) {
         this.httpPipeline = httpPipeline;
         return this;
     }
 
     /**
-     * Sets The HTTP client used to send the request.
+     * Sets the HTTP client to use for sending and receiving requests to and from the service.
      *
-     * @param httpClient the httpClient value.
-     * @return the ContainerRegistryBuilder.
+     * @param httpClient The HTTP client to use for requests.
+     * @return The updated {@link ContainerRegistryClientBuilder} object.
      */
     public ContainerRegistryClientBuilder httpClient(HttpClient httpClient) {
         this.httpClient = httpClient;
         return this;
     }
 
     /**
-     * Sets the client options such as application ID and custom headers to set on a request.
+     * Sets the {@link ClientOptions} which enables various options to be set on the client. For example setting an
+     * {@code applicationId} using {@link ClientOptions#setApplicationId(String)} to configure
+     * the {@link UserAgentPolicy} for telemetry/monitoring purposes.
+     *
+     * <p>More About <a href="https://azure.github.io/azure-sdk/general_azurecore.html#telemetry-policy">Azure Core: Telemetry policy</a>
      *
-     * @param clientOptions The {@link ClientOptions}.
-     * @return The updated {@code TableClientBuilder}.
+     * @param clientOptions {@link ClientOptions}.
+     *
+     * @return the updated {@link ContainerRegistryClientBuilder} object
      */
     public ContainerRegistryClientBuilder clientOptions(ClientOptions clientOptions) {
         this.clientOptions = clientOptions;
         return this;
     }
 
     /**
-     * Sets The configuration store that is used during construction of the service client.
+     * Sets the configuration store that is used during construction of the service client.
+     *
```

Comment:
```suggestion
     * <p>
```

Missing new paragraph tag, right now this would be one big paragraph